### PR TITLE
Change `fd.define_tensor`'s argument name from `symbolic_sizes` to `shape`

### PR DESCRIPTION
**Pull request by kevinstephano**
This was originally part of a larger PR, #416, were I pulled out this change to make it easier to review.

I wanted to make the argument change from `symbolic_sizes` to `shape` because the name `shape` better matches the terminology users encounter in [`numpy`](https://numpy.org/doc/stable/reference/generated/numpy.shape.html).  Pytorch uses the term `size` in the singular form to describe shape but the Tensor class has a [`shape` property](https://pytorch.org/tutorials/beginner/introyt/tensors_deeper_tutorial.html#tensor-shapes).  Also, it is not the case anymore that the shape can only be symbolic whereas the name only implies it is symbolic.  Here is the comment from the python bindings:

```
      // This version of define_tensor is the canonical version
      // that displays the values as they are passed to the IR's
      // TensorViewBuilder.
      // Each dimension can be of value:
      // -1 : Symbolic for Dynamic usage
      //  0 : Zero-element
      //  1 : Broadcast
      // >1 : Static size
      // NOTE: A Tensor defined for dynamic shape usage should only
      // contain either symbolic(-1) or broadcast(1) defined dimensions.
```

This an example of the change:

Old:

```
t0 = fd.define_tensor(symbolic_sizes=[4, 4], contiguity=[True, True], dtype=DataType.Float)
```
New:
```
t0 = fd.define_tensor(shape=[4, 4], contiguity=[True, True], dtype=DataType.Float)
```

This PR does the following:
1. Changes `fd.define_tensors` argument from `symbolic_sizes` to `shape`.
2. Changes appropriate files to use the argument name.
3. Fixed the README.md in `nvfuser/README.md` that used `contiguous` instead of `contiguity` in the documentation that was previously missed.
4. Incremented version number to `0.0.17`.

### CODE REVIEW

**Comment by kevinstephano** [comment]
!build

**Comment by kevinstephano** [comment]
!build

**Review by jacobhinkle — APPROVED** [review]
Good change. LGTM

**Review by jjsjann123 — APPROVED** [review]
LGTM

File: nvfuser/README.md [hunk 1]
```diff
@@ -72,7 +72,7 @@ t0 = fd.define_tensor(sizes=[2, 4, 6], strides=[24, 6, 1], dtype=DataType.Half)
 The list of symbolic sizes defines the number of dimensions and `-1` is given for each dimension unless it is a broadcast dimension that is defined with a `1`.  The contiguity information is viewed from right to left.  A `True` definition indicates the current dimension is contiguous with the dimension to its right.
```
Comment (rdspring1) [inline nvfuser/README.md:72]: What happens if `None` is passed as a contiguity value?

Comment (rdspring1) [reply nvfuser/README.md:72]: I think including this comment in the Readme would be helpful.

```
Each dimension can be of value:
 * -1 : Symbolic for Dynamic usage
 *  0 : Zero-element
 *  1 : Broadcast
 * >1 : Static size
NOTE: A Tensor defined for dynamic shape usage should only contain either symbolic(-1) or broadcast(1) 
defined dimensions.
```

**Pull request merged**